fimpy/pipeline/alignment.py

```diff
@@ -154,3 +156,116 @@ def apply_shifts(dataset, output_dir=None, block_size=120, n_jobs=10, verbose=Fa
         for i_block, (_, new_block) in enumerate(new_dataset.slices(as_tuples=True))
     )
     return new_dataset.finalize()
+
+
+def align_2p_volume(
+    dataset,
+    output_dir=None,
+    reference=None,
+    n_frames_ref=10,
+    across_planes=None,
+    prefilter_sigma=3.3,
+    upsample_factor=10,
+    max_shift=15,
+    n_jobs=20,
+    verbose=True,
+):
+    """ Function for complete alignment of two-photon, planar acquired stack
+
+    :param dataset: input H5Dataset
+    :param output_dir: optional, output destination directory, subdirectory aligned will appear
+    :param reference: optional, reference to align to
+    :param n_frames_ref: number of frames to take as reference mean, if reference is being calculated
+    :param across_planes: bool, True by default if reference is not provided, whether to align across planes
+    :param prefilter_sigma: feature size to filter for better alignment. if < 0 no filtering will take place
+    :param upsample_factor: granularity of subpixel shift
+    :param max_shift: maximum shift allowed
+    :param n_jobs: number of parallel jobs
+    :return: reference to align dataset
+    """
+
+    # prepare the destination
+    new_dataset = EmptySplitDataset(
+        root=output_dir or dataset.root.parent,
+        name="aligned",
+        shape_full=dataset.shape,
+        shape_block=(dataset.shape_block[0], 1) + dataset.shape_block[2:],
+    )
+
+    if verbose:
```

Comment:
we should replace all of those `verbose` with the logging module (can be done in a later PR)